File: dataset/output/GEMINI_gemini-1.5-pro-001/Task30_GEMINI_gemini-1.5-pro-001.java
import java.util.Arrays;

class Solution {
    public static String longest(String s1, String s2) {
        boolean[] seen = new boolean[26];
        StringBuilder sb = new StringBuilder();
        for (char c : (s1 + s2).toCharArray()) {
            int index = c - 'a';
            if (!seen[index]) {
                sb.append(c);
                seen[index] = true;
            }
        }
        char[] arr = sb.toString().toCharArray();
        Arrays.sort(arr);
        return new String(arr);
    }

    public static void main(String[] args) {
        System.out.println(longest("xyaabbbccccdefww", "xxxxyyyyabklmopq"));
        System.out.println(longest("abcdefghijklmnopqrstuvwxyz", "abcdefghijklmnopqrstuvwxyz"));
    }
}

File: dataset/output/GEMINI_gemini-1.5-pro-001/Task30_GEMINI_gemini-1.5-pro-001.cpp
#include <string>
#include <set>
#include <algorithm>

using namespace std;

string longest(const string& s1, const string& s2) {
    set<char> seen;
    string result;
    for (char c : s1 + s2) {
        if (seen.insert(c).second) {
            result += c;
        }
    }
    sort(result.begin(), result.end());
    return result;
}